Misc/NEWS.d/next/Core and Builtins/2019-09-12-16-36-16.bpo-38106.4pApn7.rst

```diff
@@ -0,0 +1,2 @@
+Fix race in PyThread_release_lock that was leading to memory corruption and
+deadlocks.
```

Comment:
Oh! I just created a very similar local change to prepare a pull request... and then I saw that you created a PR! Oops.

I wrote a different description for the change:

```
    bpo-38106: Fix a race condition in pthread locks
    
    Fix a race condition in lock of the pthread implementation using
    mutex with condition variable. Signal condition variables with the
    mutex held. Destroy condition variables before their mutexes.
```

Please mention which exact implementation is changed. We do have 2 implementations for pthread :-)